controllers/storagecluster/onboarding_validation_keys_generator_job.go

```diff
@@ -0,0 +1,109 @@
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	"k8s.io/klog/v2"
+)
+
+const (
+	onboardingValidationKeysGeneratorImage   = "ONBOARDING_VALIDATION_KEYS_GENERATOR_IMAGE"
+	onboardingValidationKeysGeneratorJobName = "onboarding-validation-keys-generator"
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+)
+
+var onboardingJobSpec = batchv1.JobSpec{
+	// Eligible to delete automatically when job finishes
+	TTLSecondsAfterFinished: ptr.To(int32(0)),
+	Template: corev1.PodTemplateSpec{
+		Spec: corev1.PodSpec{
+			RestartPolicy:      corev1.RestartPolicyOnFailure,
+			ServiceAccountName: onboardingValidationKeysGeneratorJobName,
+			Containers: []corev1.Container{
+				{
+					Name:    onboardingValidationKeysGeneratorJobName,
+					Image:   os.Getenv(onboardingValidationKeysGeneratorImage),
+					Command: []string{"/usr/local/bin/onboarding-validation-keys-gen"},
+					Env: []corev1.EnvVar{
+						{
+							Name:  util.OperatorNamespaceEnvVar,
+							Value: os.Getenv(util.OperatorNamespaceEnvVar),
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+type onboardingValidationKeysGeneratorJob struct{}
+
+var _ resourceManager = &onboardingValidationKeysGeneratorJob{}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if !instance.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("Spec.AllowRemoteStorageConsumers is disabled, skipping onboarding validation key generator job creation")
+		return reconcile.Result{}, nil
+	}
+
+	if res, err := o.createJob(r, instance); err != nil {
+		return reconcile.Result{}, err
+	} else if !res.IsZero() {
+		return res, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (o *onboardingValidationKeysGeneratorJob) createJob(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	var err error
+	r.Log.Info("Spec.AllowRemoteStorageConsumers is enabled. test ritesh Creating Onboarding validation key generator job")
+	if os.Getenv(onboardingValidationKeysGeneratorImage) == "" {
+		err = fmt.Errorf("OnboardingSecretGeneratorImage env var is not set")
+		r.Log.Error(err, "No value set for env variable")
+		return reconcile.Result{}, err
+	}
+	publicKeySecret := &corev1.Secret{}
+	publicKeySecret.Name = onboardingValidationPublicKeySecretName
+	publicKeySecret.Namespace = instance.Namespace
+	actualSecret := &corev1.Secret{}
+	// Creating the job only if public is not found
+	err = r.Client.Get(r.ctx, client.ObjectKeyFromObject(publicKeySecret), actualSecret)
+	if errors.IsNotFound(err) {
+		onboardingSecretGeneratorJob := &batchv1.Job{}
+		onboardingSecretGeneratorJob.Name = onboardingValidationKeysGeneratorJobName
+		onboardingSecretGeneratorJob.Namespace = instance.Namespace
+		onboardingSecretGeneratorJob.Spec = *onboardingJobSpec.DeepCopy()
+		err = r.Client.Create(r.ctx, onboardingSecretGeneratorJob)
+		if err != nil {
```

Comment:
should you skip checking `AlreadyExist` error as you may get many reconciles before storagecluster gets stable or other route is set the job failure policy to `Never` and recreate the job every time which is wasteful of resources i guess.